### PR TITLE
nfts sort direction 

### PR DESCRIPTION
**Pull request by greg-schrammel**
Fixes APP-####

## What changed (plus any additional context for devs)

adds options to order asc/desc to collectibles 

## Screen recordings / screenshots

https://github.com/user-attachments/assets/2421773b-f118-45a1-9a9d-7ff31a3ad6f3

### CODE REVIEW

**Comment by brunobar79** [comment]
Launch in [simulator](http://localhost:29070/install/ios?virtual=https://app-team.p.rainbow.me/gregs/app-1756-sorting-nfts-on-android-is-broken/433c00cf63fbfc2d4c40c47165841242d010a306.app.zip) or [device](http://localhost:29070/install/ios?physical=https://app-team.p.rainbow.me/gregs/app-1756-sorting-nfts-on-android-is-broken/433c00cf63fbfc2d4c40c47165841242d010a306.ipa) for 433c00cf63fbfc2d4c40c47165841242d010a306

File: src/components/cards/MintsCard/Menu.tsx [hunk 1]
```diff
@@ -27,7 +27,7 @@ export function Menu() {
         menuState: filter === MintsFilter.Paid ? 'on' : 'off',
       },
     ],
-  };
+  } as const;
```
Comment (walmat) [inline src/components/cards/MintsCard/Menu.tsx:30]: is this necessary?

File: src/components/change-wallet/AddressRow.tsx [hunk 2]
```diff
@@ -166,18 +167,15 @@ export default function AddressRow({ contextMenuActions, data, editMode, onPress
               iconValue: 'bell.fill',
             },
           },
-        ]
+        ] as const)
```
Comment (walmat) [inline src/components/change-wallet/AddressRow.tsx:170]: is this necessary?

File: src/screens/transaction-details/components/TransactionDetailsStatusActionsAndTimestampSection.tsx [hunk 3]
```diff
@@ -36,36 +35,37 @@ export const TransactionDetailsStatusActionsAndTimestampSection: React.FC<Props>
...
-                  iconType: 'SYSTEM',
-                  iconValue: 'xmark.circle',
+              ]
+            : []),
+          ...(canBeCancelled
+            ? [
+                {
+                  actionKey: 'cancel',
+                  actionTitle: i18n.t(i18n.l.transaction_details.actions_menu.cancel),
+                  menuAttributes: ['destructive' as const],
```
Comment (walmat) [inline src/screens/transaction-details/components/TransactionDetailsStatusActionsAndTimestampSection.tsx:59]: is this necessary to cast as `const`

File: src/__swaps__/screens/Swap/components/GasButton.tsx [hunk 4]
```diff
@@ -184,14 +181,11 @@ const GasMenu = ({ backToReview = false, children }: { backToReview?: boolean; c
...
         <ContextMenuButton
-          activeOpacity={0}
           enableContextMenu
-          isAnchoredToRight
           isMenuPrimaryAction
           menuConfig={menuConfig}
           onPressMenuItem={handlePressMenuItem}
           useActionSheetFallback={false}
-          wrapNativeComponent={false}
         >
```
Comment (walmat) [inline src/__swaps__/screens/Swap/components/GasButton.tsx:189]: why were these props removed?

**Comment by walmat** [comment]
Screenshot 2024-09-12 at 1 30 02 PM

On Android the context menu is going offscreen a bit. I'd say this is low priority since I'm reworking context menus anyway, so dw about it, but just wanted to flag this.

File: src/__swaps__/screens/Swap/components/GasButton.tsx (hunk 4 above)
Comment (greg-schrammel) [reply src/__swaps__/screens/Swap/components/GasButton.tsx:189]: added `isAnchoredToRight` back didn't notice it's used in the `.android`, the others are already the default inside `<ContextMenuButton />`

File: src/screens/transaction-details/components/TransactionDetailsStatusActionsAndTimestampSection.tsx (hunk 3 above)
Comment (greg-schrammel) [reply src/screens/transaction-details/components/TransactionDetailsStatusActionsAndTimestampSection.tsx:59]: yes, it considers 'destructive' as `string` but the type wants the it as the string literal type `destructive`, same stuff for the others `as const` or `satisfies` added, I expanded the type a lil bit, but in these cases opted to keep it like this so we have better auto complete

File: src/__swaps__/screens/Swap/components/GasButton.tsx (hunk 4 above)
Comment (greg-schrammel) [reply src/__swaps__/screens/Swap/components/GasButton.tsx:189]: btw seems like this GasButton menu is not working at all in android? idk if it's just my simulator

**Comment by brunobar79** [comment]
Launch in [simulator](http://localhost:29070/install/ios?virtual=https://app-team.p.rainbow.me/gregs/app-1756-sorting-nfts-on-android-is-broken/f3f5f1075155ea7132a4d78628ed16ecdd6a5f45.app.zip) or [device](http://localhost:29070/install/ios?physical=https://app-team.p.rainbow.me/gregs/app-1756-sorting-nfts-on-android-is-broken/f3f5f1075155ea7132a4d78628ed16ecdd6a5f45.ipa) for f3f5f1075155ea7132a4d78628ed16ecdd6a5f45

**Review by benisgold — COMMENTED** [review]
idk if this is a bug or not, but i'd expect the "grayed out" option to be the one that's already selected which is currently not the case

https://github.com/user-attachments/assets/757a3e90-1120-449a-88b2-a4b35a187324

**Comment by brunobar79** [comment]
Launch in [simulator](http://localhost:29070/install/ios?virtual=https://app-team.p.rainbow.me/gregs/app-1756-sorting-nfts-on-android-is-broken/06ddea3c3508d4c6d6cfbff7d6ec8f89685be196.app.zip) or [device](http://localhost:29070/install/ios?physical=https://app-team.p.rainbow.me/gregs/app-1756-sorting-nfts-on-android-is-broken/06ddea3c3508d4c6d6cfbff7d6ec8f89685be196.ipa) for 06ddea3c3508d4c6d6cfbff7d6ec8f89685be196

**Review by benisgold — APPROVED** [review]
is it possible to prevent the dismissal of the context menu if the button is disabled (ie if the sort direction clicked is the one already selected)? fine if not but that would be better

**Comment by greg-schrammel** [comment]
> is it possible to prevent the dismissal of the context menu if the button is disabled (ie if the sort direction clicked is the one already selected)? fine if not but that would be better

tried but couldn't make it work, believe we can't 😔

**Comment by brunobar79** [comment]
Launch in [simulator](http://localhost:29070/install/ios?virtual=https://app-team.p.rainbow.me/gregs/app-1756-sorting-nfts-on-android-is-broken/f2bede09f741a0db99b37f463456fbf62fa75c21.app.zip) or [device](http://localhost:29070/install/ios?physical=https://app-team.p.rainbow.me/gregs/app-1756-sorting-nfts-on-android-is-broken/f2bede09f741a0db99b37f463456fbf62fa75c21.ipa) for f2bede09f741a0db99b37f463456fbf62fa75c21

**Review by walmat — APPROVED** [review]

**Pull request merged**